addons/hr_timesheet/wizard/project_task_create_timesheet.py

```diff
@@ -20,16 +20,28 @@ class ProjectTaskCreateTimesheet(models.TransientModel):
     )
 
     def save_timesheet(self):
-        values = {
-            'task_id': self.task_id.id,
-            'project_id': self.task_id.project_id.id,
-            'date': datetime.now(),
-            'name': self.description,
-            'user_id': self.env.uid,
-            'unit_amount': self.time_spent,
-        }
-        self.task_id.user_timer_id.unlink()
-        return self.env['account.analytic.line'].create(values)
+        timer = self.env['timer.timer'].search([
+            ('user_id', '=', self.env.user.id),
+            ('res_model', '=', 'account.analytic.line'),
+            ('res_id', 'in', self.task_id.timesheet_ids.ids),
+            ('timer_start', '!=', False),
+            ('timer_pause', '=', False)
+        ], limit=1)
+        if timer:
+            analytic_line = self.env[timer.res_model].search([('id', '=', timer.res_id)])
```

Comment:
Don't hesitate to add a comment to explain the reason behind your addition. Basically, just your commit text to avoid having to read the commit to understand at a glance what you are doing.

Reply:
@arj-odoo done